### PR TITLE
2150 removing observed data

### LINKED ISSUE #2150: Removing observed data entry from simulation - do not show a warning
Removing observed data from the "Observed data" tab of a simulation shows a warning every time:

image

This is annoying and unnecessary. Add an option "Do not ask again".

### PR DESCRIPTION
**Pull request by peterdew**
Fixes #2150 

# Description
To prevent showing the same confirmationmessages over and over again, users can now suppress it when deleting observed data from a simulation until next restart.

For easily expanding this behaviour to more confirmation-messages there is confirmationMessage-type added, which has a checkbox to suppress further confirmations. This accepts a handler which executes when the checkbox is clicked and the action is confirmed. 

There is a service added, the ConfirmationManager, to hold and manage a collection of confirmations to suppress. 

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):
2024-03-19 14_47_46-MoBi® 12 0 0+53e21410768fb2c5dbf152c5e74e0a82e42d1bfa

# Questions (if appropriate):

### CODE REVIEW

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs
Comment (Yuri05) [inline tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs:1]: please rename to ConfirmationManager**Specs**.cs

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+{
+   public abstract class concern_for_ConfirmationManager : ContextSpecification<IConfirmationManager>
+   {
+      protected override void Context()
+      {
+         sut = new ConfirmationManager();
+      }
+   }
+
+   internal class When_add_a_suppression_flag : concern_for_ConfirmationManager
```
Comment (Yuri05) [inline tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs:14]: please rename to When_add**ing**_a_suppression_flag

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   internal class When_there_is_no_suppression_flag : concern_for_ConfirmationManager
+   {
+      [Observation]
+      public void should_deny_suppressions()
+      {
+         sut.IsConfirmationSuppressed(ConfirmationFlags.ObservedDataEntryRemoved).ShouldBeFalse();
+      }
+   }
+
+    internal class When_add_a_flag_that_already_exists : concern_for_ConfirmationManager
```
Comment (Yuri05) [inline tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs:36]: pls rename to When_add**ing**_a_flag_that_already_exists

File: src/OSPSuite.UI/Services/DialogCreator.cs [hunk 3]
```diff
@@ -79,24 +83,46 @@ public ViewResult MessageBoxYesNo(string message, ViewResult defaultButton = Vie
...
          }
          finally
          {
             Localizer.Active = currentLocalizer;
          }
       }
 
+      private void onMessageBoxClosed(XtraMessageBoxClosedArgs e, Action doNotShowAgain)
+      {
+         // e.Visible gets whether the user checked the 'Do not show this message again' checkbox. Checked makes e.Visible = false
```
Comment (Yuri05) [inline src/OSPSuite.UI/Services/DialogCreator.cs:119]: I don't get it. 
Why does `e.Visible` correspond to the check status of the 'Do not show this message again' checkbox?

**Review by Yuri05 — CHANGES_REQUESTED** [review]
s. comments

File: src/OSPSuite.UI/Services/DialogCreator.cs (hunk 3 above)
Comment (rwmcintosh) [reply src/OSPSuite.UI/Services/DialogCreator.cs:119]: I think you'd have to ask the DevExpress guys why it's named that

https://docs.devexpress.com/WindowsForms/DevExpress.XtraEditors.XtraMessageBoxEventArgs.Visible

Comment (Yuri05) [reply src/OSPSuite.UI/Services/DialogCreator.cs:119]: Strange.. but OK

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs (hunk 2 above)
Comment (peterdew) [reply tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs]: Fixed by last commit

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs
Comment (peterdew) [reply tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs:1]: Changed it

File: tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs (hunk 1 above)
Comment (peterdew) [reply tests/OSPSuite.Core.Tests/Services/ConfirmationManager.cs]: Done

**Review by Yuri05 — APPROVED** [review]

**Comment by PavelBal** [comment]
Hmm, this is not in the nightly MoBi? @rwmcintosh

**Comment by rwmcintosh** [comment]
Core update has not been done in a little while. I'll include an update in the changes I'm working on now. That PR should be ready today.

**Pull request merged**